src/libostree/ostree-sysroot-cleanup.c

```diff
@@ -353,34 +354,18 @@ cleanup_ref_prefix (OstreeRepo         *repo,
                     GCancellable       *cancellable,
                     GError            **error)
 {
-  gboolean ret = FALSE;
-  g_autofree char *prefix = NULL;
-  g_autoptr(GHashTable) refs = NULL;
-  GHashTableIter hashiter;
-  gpointer hashkey, hashvalue;
-
-  prefix = g_strdup_printf ("ostree/%d/%d", bootversion, subbootversion);
+  g_autofree char *prefix = g_strdup_printf ("ostree/%d/%d", bootversion, subbootversion);
 
+  g_autoptr(GHashTable) refs = NULL;
   if (!ostree_repo_list_refs_ext (repo, prefix, &refs, OSTREE_REPO_LIST_REFS_EXT_NONE, cancellable, error))
-    goto out;
-
-  if (!ostree_repo_prepare_transaction (repo, NULL, cancellable, error))
-    goto out;
+    return FALSE;
 
-  g_hash_table_iter_init (&hashiter, refs);
-  while (g_hash_table_iter_next (&hashiter, &hashkey, &hashvalue))
+  GLNX_HASH_TABLE_FOREACH (refs, const char *, ref)
     {
-      const char *ref = hashkey;
       ostree_repo_transaction_set_refspec (repo, ref, NULL);
```

Comment:
Actually, can't we just use `ostree_repo_set_ref_immediate()` for these?

Reply:
Yep, done.  Also added tests ⬇️